### PR TITLE
Fix empty OIDC config prevents fresh application from starting

### PR DESCRIPTION
**Pull request by jamilbk**
~~Possibly due to the empty config? Added a quick fix for now.~~

Edit: See the issue now -- `openid_connect_providers` is expected to be a JSON string now. If `AUTH_OIDC` is missing from the env then the default of `[]` applies, which is an elixir list. Updated to use a default of `"{}"`

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -58,9 +58,8 @@ services:
       POSTGRES_USER: postgres
       POSTGRES_PASSWORD: postgres
       POSTGRES_DB: firezone_dev
-    # when you want to connect db with a graphic tool, uncomment ports
-    # ports:
-    #   - 5432:5432
+    ports:
```
Comment (jamilbk) [inline docker-compose.yml:61]: Helpful for debugging

Comment (princemaple) [reply docker-compose.yml:61]: Yup. I thought some of you might have non-docker instances running on the machine. Thought to avoid conflicts.

**Comment by princemaple** [comment]
Found this in the logs. Is this related? I thought it'd automatically migrate.

```
 Notice: 25_19:38:22.23938 19:38:21.368 [notice] Application fz_http exited: FzHttp.Application.start(:normal, []) returned an error: shutdown: failed to start child: FzHttp.Conf.Cache
2022-07-25_19:38:22.23942     ** (EXIT) an exception was raised:
2022-07-25_19:38:22.23943         ** (Postgrex.Error) ERROR 42P01 (undefined_table) relation "configurations" does not exist
2022-07-25_19:38:22.23943 
2022-07-25_19:38:22.23943     query: SELECT c0."id", c0."logo", c0."local_auth_enabled", c0."allow_unprivileged_device_management", c0."openid_connect_providers", c0."disable_vpn_on_oidc_error", c0."auto_create_oidc_users", c0."inserted_at", c0."updated_at" FROM "configurations" AS c0
2022-07-25_19:38:22.23943             (ecto_sql 3.8.3) lib/ecto/adapters/sql.ex:932: Ecto.Adapters.SQL.raise_sql_call_error/1
2022-07-25_19:38:22.23943             (ecto_sql 3.8.3) lib/ecto/adapters/sql.ex:847: Ecto.Adapters.SQL.execute/6
2022-07-25_19:38:22.23943             (ecto 3.8.4) lib/ecto/repo/queryable.ex:221: Ecto.Repo.Queryable.execute/4
2022-07-25_19:38:22.23944             (ecto 3.8.4) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
2022-07-25_19:38:22.23944             (ecto 3.8.4) lib/ecto/repo/queryable.ex:155: Ecto.Repo.Queryable.one!/3
2022-07-25_19:38:22.23944             (fz_http 0.5.0) lib/fz_http/conf/cache.ex:27: FzHttp.Conf.Cache.init/1
2022-07-25_19:38:22.23944             (stdlib 4.0.1) gen_server.erl:848: :gen_server.init_it/2
2022-07-25_19:38:22.23945             (stdlib 4.0.1) gen_server.erl:811: :gen_server.init_it/6
```

**Comment by jamilbk** [comment]
> Found this in the logs. Is this related? I thought it'd automatically migrate.
> 
> ```
>  Notice: 25_19:38:22.23938 19:38:21.368 [notice] Application fz_http exited: FzHttp.Application.start(:normal, []) returned an error: shutdown: failed to start child: FzHttp.Conf.Cache
> 2022-07-25_19:38:22.23942     ** (EXIT) an exception was raised:
> 2022-07-25_19:38:22.23943         ** (Postgrex.Error) ERROR 42P01 (undefined_table) relation "configurations" does not exist
> 2022-07-25_19:38:22.23943 
> 2022-07-25_19:38:22.23943     query: SELECT c0."id", c0."logo", c0."local_auth_enabled", c0."allow_unprivileged_device_management", c0."openid_connect_providers", c0."disable_vpn_on_oidc_error", c0."auto_create_oidc_users", c0."inserted_at", c0."updated_at" FROM "configurations" AS c0
> 2022-07-25_19:38:22.23943             (ecto_sql 3.8.3) lib/ecto/adapters/sql.ex:932: Ecto.Adapters.SQL.raise_sql_call_error/1
> 2022-07-25_19:38:22.23943             (ecto_sql 3.8.3) lib/ecto/adapters/sql.ex:847: Ecto.Adapters.SQL.execute/6
> 2022-07-25_19:38:22.23943             (ecto 3.8.4) lib/ecto/repo/queryable.ex:221: Ecto.Repo.Queryable.execute/4
> 2022-07-25_19:38:22.23944             (ecto 3.8.4) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
> 2022-07-25_19:38:22.23944             (ecto 3.8.4) lib/ecto/repo/queryable.ex:155: Ecto.Repo.Queryable.one!/3
> 2022-07-25_19:38:22.23944             (fz_http 0.5.0) lib/fz_http/conf/cache.ex:27: FzHttp.Conf.Cache.init/1
> 2022-07-25_19:38:22.23944             (stdlib 4.0.1) gen_server.erl:848: :gen_server.init_it/2
> 2022-07-25_19:38:22.23945             (stdlib 4.0.1) gen_server.erl:811: :gen_server.init_it/6
> ```

Yeah but I think that is being cause by this timeout:

```
        
        ================================================================================
        Error executing action `run` on resource 'ruby_block[restart_log_service]'
        ================================================================================
        
        Mixlib::ShellOut::ShellCommandFailed
        ------------------------------------
        Expected process to exit with [0], but received '1'
        ---- Begin output of /opt/firezone/embedded/bin/sv restart /opt/firezone/service/wireguard/log ----
        STDOUT: timeout: run: /opt/firezone/service/wireguard/log: (pid 14411) 31s, got TERM
        STDERR: 
        ---- End output of /opt/firezone/embedded/bin/sv restart /opt/firezone/service/wireguard/log ----
        Ran /opt/firezone/embedded/bin/sv restart /opt/firezone/service/wireguard/log returned 1
```

The weird thing is we haven't touched that recipe in a long time. I'm investigating what the issue could be...

**Review by princemaple — APPROVED** [review]
🥳

**Comment by jamilbk** [comment]
> 🥳

🎉 🎉 🎉

**Pull request merged**